src/api/v2/v2AzureResourcesApi.ts

```diff
@@ -89,6 +93,14 @@ export interface WrappedResourceModel {
  * The interface that resource resolvers must implement
  */
 export interface BranchDataProvider<TResource extends ResourceBase, TModel extends ResourceModelBase> extends vscode.TreeDataProvider<TModel> {
+    /**
+     * Get the children of `element` or root if no element is passed.
+     *
+     * @param element The element from which the provider gets children. Unlike a traditional TreeDataProvider, this will never be `undefined`.
+     * @return Children of `element` or root if no element is passed.
+     */
+    getChildren(element: TModel): vscode.ProviderResult<TModel[]>;
+
     /**
```

Comment:
It's not necessarily required, but do we want to do this as well, to indicate that a BDP should _not_ implement `getParent()`, because it's not necessary for them to do?

```typescript
    getParent?: never;
```

Reply:
Added it.